compiler/ballerina-tools-api/src/main/java/io/ballerina/tools/text/TextDocument.java

```diff
@@ -50,6 +53,14 @@ public int textPositionFrom(LinePosition linePosition) {
         return lines().textPositionFrom(linePosition);
     }
 
+    public List<String> textLines() {
+        if (lineMap != null) {
+            return lineMap.textLines();
+        }
+        lineMap = populateTextLineMap();
+        return Collections.unmodifiableList(lineMap.textLines());
```

Comment:
We can make the `lineMap.textLines()` return an unmodifiable list

Reply:
Changed it so that linemap class returns unmodifiable list